base/string_utils.hpp

```diff
@@ -306,6 +306,11 @@ void Tokenize(string const & str, char const * delims, TFunctor && f)
   }
 }
 
+/// Splits a string by the delimiter, keeps empty parts, on an empty string returns an empty vector.
+/// Does not support quoted columns, newlines in columns and escaped quotes.
+/// @return false if the line is empty or number of columns differs from |columns|.
+bool ParseCSVRow(string const & s, vector<string> & target, char const delimiter = ',', size_t const columns = 0);
```

Comment:
I guess the API must be as generic as possible - why do you need to pass columns here? Why not to check (if you need it) number of columns at the call place?